### PR TITLE
Improve instrumentation tests

### LINKED ISSUE #496: Improve tests
We should add a test to cover https://github.com/tonerdo/coverlet/pull/495

cc: @ViktorHofer @tonerdo

### PR DESCRIPTION
**Pull request by daveMueller**
Closes #496

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
Thanks @daveMueller I was on vacation I'll take a look asap!

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 1]
```diff
@@ -42,10 +43,27 @@ public void TestCoreLibInstrumentation()
...
 
-        [Fact]
-        public void TestInstrument()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void TestInstrument(bool singleHit)
         {
-            var instrumenterTest = CreateInstrumentor();
+            var instrumenterTest = CreateInstrumentor(singleHit:singleHit);
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]:
```suggestion
            var instrumenterTest = CreateInstrumentor(singleHit: singleHit);
```

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 2]
```diff
@@ -56,9 +74,26 @@ public void TestInstrument()
         }
 
         [Fact]
-        public void TestInstrumentCoreLib()
+        public void TestInstrumentWithExcludedFiles()
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: I would remove this test...at the moment there is a bug in implementation...this PR should fix https://github.com/tonerdo/coverlet/pull/524
After that we won't need anymore to search file on parent and we can simply pass glob pattern.

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 3]
```diff
@@ -187,6 +223,8 @@ class InstrumenterTest
             public string Identifier { get; set; }
 
             public DirectoryInfo Directory { get; set; }
+
+            public Mock<ILogger> Logger { get; set; }
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: I don't see any reference to this prop...it's a typo or I'm missing something?

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 4]
```diff
@@ -253,5 +291,14 @@ public void SkipEmbeddedPpdbWithoutLocalSource()
             loggerMock.VerifyNoOtherCalls();
         }
 
+        [Fact]
+        public void CanInstrumentReturnsFalseAndLogsCatchedException()
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: I would rename `TestInstrument_MissingModule`

File: test/coverlet.core.tests/Samples/Samples.cs [hunk 5]
```diff
@@ -4,6 +4,8 @@
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
+using System.Reflection;
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Samples/Samples.cs]: These using seem unuseful...can we remove?

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs (hunk 1 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs (hunk 2 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs (hunk 3 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done. Sorry missed to delete the property before pushing.

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs (hunk 4 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done, renamed.

File: test/coverlet.core.tests/Samples/Samples.cs (hunk 5 above)
Comment (daveMueller) [reply test/coverlet.core.tests/Samples/Samples.cs]: Done

File: test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs [hunk 6]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.IO;
+using System.IO.Compression;
```
Comment (MarcoRossignoli) [inline test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]:
```suggestion
```

Comment (daveMueller) [reply test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs]: Done

**Pull request merged**